modules/openthread/platform/infra_if.c

```diff
@@ -35,6 +35,7 @@ static uint32_t ail_iface_index;
 static struct net_icmp_ctx ra_ctx;
 static struct net_icmp_ctx rs_ctx;
 static struct net_icmp_ctx na_ctx;
+static struct in6_addr mcast_addr;
```

Comment:
Use the namespaced types

```suggestion
static struct net_in6_addr mcast_addr;
```